src/transformers/hf_argparser.py

```diff
@@ -0,0 +1,110 @@
+import dataclasses
+from argparse import ArgumentParser
+from enum import Enum
+from typing import Any, Iterable, NewType, Tuple, Union
+
+
+DataClass = NewType("DataClass", Any)
+DataClassType = NewType("DataClassType", Any)
+
+
+class HfArgumentParser(ArgumentParser):
+    """
+    This subclass of `argparse.ArgumentParser` uses type hints on dataclasses
+    to generate arguments.
+    """
+
+    def __init__(self, obj: Union[None, DataClass, DataClassType] = None):
```

Comment:
I would let it accept a list of DataClass as well store these in a `self.dataclasse_types` list (in the `add_dataclass_arguments ` method) to be able to generate the dataclasses when parsing like in the class method `parse_into_dataclasses`.